machine/v1beta1/types_awsprovider.go

```diff
@@ -95,6 +95,18 @@ type AWSMachineProviderConfig struct {
 	// The field size should be greater than 0 and the field input must start with cr-***
 	// +optional
 	CapacityReservationID string `json:"capacityReservationId"`
+	// marketType specifies the type of market for the EC2 instance.
+	// Valid values are OnDemand, Spot, CapacityBlock and omitted.
+	//
+	// Defaults to OnDemand.
+	// When SpotMarketOptions is provided, the marketType defaults to "Spot".
+	//
+	// When set to OnDemand the instance runs as a standard OnDemand instance.
+	// When set to Spot the instance runs as a Spot instance.
+	// When set to CapacityBlock the instance utilizes pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
+	// If this value is selected, CapacityReservationID must be specified to identify the target reservation.
```

Comment:
Use the serialized version when referring to other fields
```suggestion
	// If this value is selected, capacityReservationID must be specified to identify the target reservation.
```